packages/app/pages/course/[cid]/insights.tsx

```diff
@@ -1,40 +1,84 @@
-import { InfoCircleOutlined } from "@ant-design/icons";
+import React, { ReactElement, useState } from "react";
 import { API } from "@koh/api-client";
-import { InsightDisplay } from "@koh/common";
-import { Card, Tooltip } from "antd";
+import useSWR from "swr";
+import { Tooltip, Card, Space, Drawer, Button } from "antd";
 import { CardSize } from "antd/lib/card";
+import { CloseSquareOutlined, InfoCircleOutlined } from "@ant-design/icons";
 import { useRouter } from "next/router";
-import React, { ReactElement } from "react";
-import useSWR from "swr";
 import { StandardPageContainer } from "../../../components/common/PageContainer";
+import { InsightDisplay } from "@koh/common";
+import NavBar from "../../../components/Nav/NavBar";
 import SimpleChartComponent from "../../../components/Insights/components/SimpleChartComponent";
 import SimpleDisplayComponent from "../../../components/Insights/components/SimpleDisplayComponent";
+import InsightsDisplayOptions from "../../../components/Insights/components/InsightsDisplayOptions";
 
 export default function Insights(): ReactElement {
-  // TODO: In the future this will come from the users specific insights that want to see
-  const insights = ["TotalQuestionsAsked", "TotalStudents"];
+  const router = useRouter();
+  const { cid } = router.query;
+  const { data: profile, error, mutate } = useSWR(`api/v1/profile`, async () =>
```

Comment:
food for thought, but not this pr, should we move `useProfile` to also export `data, error`, and `mutate`? It seemsl ike we need it a lot these days

Reply:
Yeah I totally agree.